tasks/configure.yml

```diff
@@ -12,15 +12,22 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 ---
-- include: configure-api-key.yml
+- include: configure-agent.yml
 
 - include: configure-plugins.yml
 
-- name: Ensure collectd is running.
-  service: name={{ stackdriver_collectd_service }} state=started enabled=yes
+- name: "Ensure collectd is {% if not stackdriver_enabled %}not {% endif %}running."
+  service:
+    name: "{{ stackdriver_collectd_service }}"
+    state: "{% if stackdriver_enabled %}started{% else %}stopped{% endif %}"
+    enabled: "{% if stackdriver_enabled %}yes{% else %}no{% endif %}"
 
-- name: Ensure extractor is running.
-  service: name={{ stackdriver_extractor_service }} state=started enabled=yes
+- name: "Ensure extractor is {% if not (stackdriver_enabled and not stackdriver_use_gcm) %}not {% endif %}running."
```

Comment:
`{% if stackdriver_use_gcm or not stackdriver_enabled %}` (simpler and consistent with below).